.github/workflows/parallel.yml

```diff
@@ -69,6 +69,13 @@ jobs:
     steps:
     - name: Set /etc/hosts mapping
       run: sudo hostname "action-host" | sudo echo "127.0.0.1 action-host" | sudo tee -a /etc/hosts
+    - name: Cache local Maven repository
```

Comment:
In line 86, we are already caching maven using the default key. 
If you wish to customize key, please add `matric.profile` in key name.
Also, it may be better to cache maven after setup jdk.

Reply:
Emm... If this cache is enabled, but why it always download packages every time, this makes me confused.

Reply:
Because they are using the same key, if checkstyle finishes first, only dependecies for checkstyle will be cached.

Reply:
You can still change this PR to use a different key for each profile.
However, I've tried this before, there weren't much difference in total CI execution time.